src/geodesics.cpp

```diff
@@ -99,49 +101,217 @@ class HeatGeodesicSolver {
 };
 
 
+std::tuple<std::vector<compas::RowMatrixXd>, std::vector<compas::RowMatrixXi>>
+pmp_geodesic_isolines_split(
+    Eigen::Ref<compas::RowMatrixXd> vertices,
+    Eigen::Ref<compas::RowMatrixXi> faces,
+    const std::vector<int>& sources,
+    const std::vector<double>& isovalues)
+{
+    using vertex_descriptor = boost::graph_traits<compas::Mesh>::vertex_descriptor;
+    using edge_descriptor = boost::graph_traits<compas::Mesh>::edge_descriptor;
+
+    // Convert input to CGAL mesh
+    compas::Mesh mesh = compas::mesh_from_vertices_and_faces(vertices, faces);
+    int n_vertices = mesh.number_of_vertices();
+
+    // Create property map for distances
+    auto distance_pmap = mesh.add_property_map<vertex_descriptor, double>("v:distance", 0.0).first;
+
+    // Create heat method solver and compute geodesic distances
+    HeatMethod hm(mesh);
+    for (int src : sources) {
+        if (src >= 0 && src < n_vertices) {
+            hm.add_source(vertex_descriptor(src));
+        }
+    }
+    hm.estimate_geodesic_distances(distance_pmap);
+
+    // Create edge constraint map to mark isoline edges
+    auto ecm = mesh.add_property_map<edge_descriptor, bool>("e:is_constrained", false).first;
+
+    // Refine mesh along each isovalue
```

Comment:
@petrasvestartas this is absolutely killer, the `idt` / mesh refinement feature, amazing!